Java/core/src/main/java/com/amazon/randomcutforest/store/IPointStoreView.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amazon.randomcutforest.store;
+
+/**
+ * A view of the PointStore that forces a read only access to the store.
+ */
+public interface IPointStoreView<P> {
+    int getDimensions();
+
+    int getCapacity();
+
+    boolean pointEquals(int index, P point);
+
+    P get(int index);
+
+    void incrementRefCount(int index);
+
+    void decrementRefCount(int index);
```

Comment:
these two methods do not appear a read only view

Reply:
Indeed!